### PR TITLE
mosquitto: Fix compilation without deprecetated OpenSSL APIs

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

Maintainer: @karlp 
Compile tested: ramips

### CODE REVIEW

**Comment by karlp** [comment]
How does this relate to https://github.com/eclipse/mosquitto/pull/1034 or any other patches upstream?

File: net/mosquitto/Makefile [hunk 1]
```diff
@@ -146,7 +147,7 @@ endef
 define Package/libmosquittopp
     SECTION:=libs
     CATEGORY:=Libraries
-    DEPENDS:=libmosquitto +libstdcpp
+    DEPENDS:=libmosquitto $(CXX_DEPENDS)
```
Comment (karlp) [inline net/mosquitto/Makefile]: this _appears_ to be completely unrelated to "compile without openssl deprecated apis"

File: net/mosquitto/patches/010-openssl-deprecated.patch [hunk 2]
```diff
@@ -0,0 +1,46 @@
+--- a/lib/net_mosq.c
++++ b/lib/net_mosq.c
+@@ -87,9 +87,11 @@ int net__init(void)
+ #endif
+ 
+ #ifdef WITH_TLS
++#if OPENSSL_VERSION_NUMBER < 0x10100000L
```
Comment (karlp) [inline net/mosquitto/patches/010-openssl-deprecated.patch]: You can have lower versions with different deprecations, are you sure you shouldbn't be using the openssl compat defines? not the version defines?

Comment (neheb) [reply net/mosquitto/patches/010-openssl-deprecated.patch]: Fairly. wget currently breaks because it uses it.

File: net/mosquitto/Makefile (hunk 1 above)
Comment (neheb) [reply net/mosquitto/Makefile]: I added a note in the commit message.

**Comment by neheb** [comment]
> How does this relate to [eclipse/mosquitto#1034](https://github.com/eclipse/mosquitto/pull/1034) or any other patches upstream?

That patch has already been merged.

**Comment by karlp** [comment]
Ok, but I see now you're touching different deprecation parts.  So, reworded, where's the upstream patch for this?

**Comment by neheb** [comment]
I have not submitted. Something about a CLA.

**Comment by karlp** [comment]
ok, well, that's just a SoB too, or at least file a ticket with the information, I'm not merging such edge patches without them having even tracked upstream.

**Comment by neheb** [comment]
Need to sign up and fill out a form just to submit a patch...

I think I'll just open an issue.

**Comment by cshoredaniel** [comment]
+run tested please

**Comment by neheb** [comment]
Fixed OpenSSL 1.0.2 as well.

@cshoredaniel there's no change in functionality. Just compilation fixes.

**Comment by cshoredaniel** [comment]
This also changes the C++ library depending on the user's configuration.

File: net/mosquitto/Makefile (hunk 1 above)
Comment (karlp) [reply net/mosquitto/Makefile]: I'm still not ever going to merge this under the title of "compile without open ssl deprecated apis"  this looks by all means a valid fix, btu it's a completely separate commit.

Comment (karlp) [reply net/mosquitto/Makefile]: also, does the $(CXX_DEPENDS) work to allow selecting the library directly? or do you now have to go and select a c++ lib first?

Comment (neheb) [reply net/mosquitto/Makefile]: Ok. I will place it under a separate commit sometime in the future.

**Comment by karlp** [comment]
I still don't see any issue filed for tracking this upstream.  I'm not carrying this forever here.

**Comment by neheb** [comment]
Done

**Comment by neheb** [comment]
Switched to upstream patch.

**Comment by neheb** [comment]
@karlp this Ok now?

**Comment by karlp** [comment]
Yes, I'm just waiting for the next release upstream.  It was meant to be friday, but will be this week instead.

**Comment by neheb** [comment]
Should I close this then?

**Comment by karlp** [comment]
No, leave it, I'm using it for my own tracking, I'll close it when the issue has been addressed.

**Comment by karlp** [comment]
this has now merged thanks.

**Pull request closed without merging**